# Homework Data Type and Structure Python

- Nama: Aulia Akbar Ramadhan
- IPK / Nilai Rapot: 3.85

Catatan:
1. Boleh google, YOLO apalagi biar ada gregetnya
2. Boleh dikerjakan sambil makan, push rank, skripsian, tesis, disertasi, nonton spiderman no way home, mengendarai pesawat terbang, terbang kebulan, tapi ga boleh dikerjakan sama mantan.
3. Kalau mau nyontek, `main rapih ya`, kalau ketauan jawabannya sama persis / similarity score > 80%, dua-duanya mendapatkan nilai 0 (Udah banyak korban di batch-batch sebelumnya)
4. Mohon ganti nama file sebagaimana mestinya, sesuai EYD dan tata penulisan yang diakui multiverse

## 1. Code Refactoring (40 pts)

> Ubah code dibawah ini menjadi bahasa python, kemudian outputnya juga harus sesuai ya.


```
scores <- 75
standard_score <- 70
lower_bound <- 5
upper_bound <- 10

if ((standard_score - lower_bound < scores)&&(standard_score + upper_bound > scores)) {
    bonus <- c(50,40,30,20,50)
    cat("Congratulation, you passed the test with score:", scores)
    cat("\n")
    cat("Your bonus ($):", sum(bonus))
} else {
    bonus <- c(10,20,30,40,5)
    cat("Unfortunately, you failed the test with score:", scores)
    cat("\n")
    cat("Your bonus ($):", sum(bonus))
}
```

Ekspektasi jawaban:

1. Jika `scores <- 75` maka:
```
Congratulation, you passed the test with score: 75
Your bonus ($): 190
```
2. Jika `scores <- 50` maka: 
```
Unfortunately, you failed the test with score: 50
Your bonus ($): 105
```

In [ ]:
# Ketik jawaban nomor 1 dibawah ini
scores = 75
standard_score = 70
lower_bound = 5
upper_bound = 10

if (standard_score - lower_bound < scores) and (standard_score + upper_bound > scores):
    bonus =[50,40,30,20,50]
    print("Congratulation, you passed the test with score:", scores)
    print("\n")
    print("Your bonus ($):", sum(bonus))
else:
    bonus= [10,20,30,40,5]
    print("Unfortunately, you failed the test with score:", scores)
    print("\n")
    print("Your bonus ($):", sum(bonus))

Congratulation, you passed the test with score: 75


Your bonus ($): 190


In [ ]:
# Ketik jawaban nomor 1 dibawah ini
scores = 50
standard_score = 70
lower_bound = 5
upper_bound = 10

if (standard_score - lower_bound < scores) and (standard_score + upper_bound > scores):
    bonus =[50,40,30,20,50]
    print("Congratulation, you passed the test with score:", scores)
    print("\n")
    print("Your bonus ($):", sum(bonus))
else:
    bonus= [10,20,30,40,5]
    print("Unfortunately, you failed the test with score:", scores)
    print("\n")
    print("Your bonus ($):", sum(bonus))

Unfortunately, you failed the test with score: 50


Your bonus ($): 105


## 2. List Operations (40 pts)

> Kamu diberikan satu set data total penjualan bulanan pada tahun 2021 dari bulan januari hingga november. Bantulah hitung berapa (1) rata-rata perbulan, (2) bulan dengan penjualan terendah, (3) bulan dengan penjualan tertinggi, dan (4) bulan dengan nilai transaksi diatas rata-rata penjualan perbulan

```
# Indeks 0 = Januari, indeks 1 = Februari, ..., indeks terakhir = November (Berurutan)
total_penjualan = [293,161,313,251,634,385,175,375,642,442,500]
```

- Ada 4 bulir soal, masing-masing 10 point jika benar

In [ ]:
# Ketik jawaban nomor 2 dibawah ini
total_penjualan=[293,161,313,251,634,385,175,375,642,442,500]
n=len(total_penjualan)
avrg=sum(total_penjualan)/n
mini=min(total_penjualan)
maks=max(total_penjualan)
print("rata-rata perbulan",avrg)
print("penjualan terendah adalah sebesar",mini)
print("penjualan tertinggi adalah sebesar",maks)
print("Total Penjualan diatas rata-rata adalah: ")
for x in total_penjualan:
  if x>sum(total_penjualan)/n:
      print(x)



rata-rata perbulan 379.1818181818182
penjualan terendah adalah sebesar 161
penjualan tertinggi adalah sebesar 642
Total Penjualan diatas rata-rata adalah: 
634
385
642
442
500


rata-rata penjualan perbulannya adalah 379.182. Bulan dengan penjualan terendah adalah bulan Februari dengan total penjualannya adalah sebesar 161. Sedangkan bulan dengan total penjualan tertinggi adalah bulan September dengan total penjualan sebesar 642. Bulan dengan penjualan diatas rata-rata adalah bulan Mei, Juni, September, Oktober, November. 

## 3. Dictionary Operations (20 pts)

> Gunakan manipulasi dictionary agar merubah variabel dictionary dari sebelum menjadi sesudah seperti dibawah

```
Before: {'name': ['thur', 'spidolman', 'dr.Weird'], 'gpa': [3.85, 3.66, 4.0], 'invoice_type': 'marketplace'}

After: {'name': ['thur', 'spidolman', 'dr.Weird', 'Iron Woman'], 'avg_gpa': 3.8366666666666664, 'invoice_types': ['marketplace', 'virtual product']}
```

- Gunakan data dari dictionary pertama (before) yang nilai-nilainya bisa diolah sehingga menjadi dictionary kedua (after)

In [ ]:
# Ketik jawaban nomor 3 dibawah ini
dicta={}
dicta={'name': ['thur', 'spidolman', 'dr.Weird'], 'gpa': [3.85, 3.66, 4.0], 'invoice_type': 'marketplace'}
dicta

{'gpa': [3.85, 3.66, 4.0],
 'invoice_type': 'marketplace',
 'name': ['thur', 'spidolman', 'dr.Weird']}

In [ ]:
dicta['avg_gpa']=sum(dicta['gpa'])/3
dicta['name']= ['thur', 'spidolman', 'dr.Weird','Iron Woman']
dicta['invoice_type']=['marketplace', 'virtual product']
del dicta['gpa']
dicta

{'avg_gpa': 3.8366666666666664,
 'invoice_type': ['marketplace', 'virtual product'],
 'name': ['thur', 'spidolman', 'dr.Weird', 'Iron Woman']}

# Semangat!